AK/Traits.h

```diff
@@ -37,7 +39,7 @@ requires(IsIntegral<T>) struct Traits<T> : public GenericTraits<T> {
 };
 
 template<typename T>
-requires(IsPointer<T>) struct Traits<T> : public GenericTraits<T> {
+requires(IsPointer<T> && !(IsSame<Detail::Decay<T>, char*> || IsSame<Detail::Decay<T>, char const*>)) struct Traits<T> : public GenericTraits<T> {
```

Comment:
I'd probably extract the `char* / char const*` check to a helper concept, since it's quite a mouthful to repeat.

Reply:
I think I found a neat abstraction for that
coming up....